Upstream Writing Tools Animations.

megangardner opened the pull request:
#### 02b1ca3eb7f20fac332f9a1d5fa3fcec3e569f25
<pre>
Upstream Writing Tools Animations.
<a href="https://bugs.webkit.org/show_bug.cgi?id=275411">https://bugs.webkit.org/show_bug.cgi?id=275411</a>
<a href="https://rdar.apple.com/129697392">rdar://129697392</a>

Reviewed by Richard Robinson.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm:
(-[WKTextIndicatorStyleEffectData initWithEffectID:type:]):
(-[WKTextIndicatorStyleEffectData effectID]):
(-[WKTextIndicatorStyleManager initWithWebViewImpl:]):
(-[WKTextIndicatorStyleManager addTextIndicatorStyleForID:withData:]):
(-[WKTextIndicatorStyleManager removeTextIndicatorStyleForID:]):
(-[WKTextIndicatorStyleManager hasActiveTextIndicatorStyle]):
(-[WKTextIndicatorStyleManager suppressTextIndicatorStyle]):
(-[WKTextIndicatorStyleManager restoreTextIndicatorStyle]):
(-[WKTextIndicatorStyleManager textPreviewsForChunk:completion:]):
(-[WKTextIndicatorStyleManager textPreviewForRect:completion:]):
(-[WKTextIndicatorStyleManager updateIsTextVisible:forChunk:completion:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/TextIndicatorStyle/TextStyleManager.swift: Added.
(currentEffect):
(delegate):
(beginEffect(for:style:)):
(endEffect(for:)):
(TextStyleManager.targetedPreview(for:)):
(TextStyleManager.updateTextChunkVisibilityForAnimation(_:visible:)):
(TextStyleManager.performReplacementAndGeneratePreview(for:effect:animation:)):
(TextStyleManager.replacementEffectDidComplete(_:)):

Canonical link: <a href="https://commits.webkit.org/280034@main">https://commits.webkit.org/280034@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e1b31ae9fcf1d7368583ce7649f518ca3bcd10c8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55538 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34861 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8005 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58522 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5968 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42483 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6167 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44722 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4100 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57567 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32768 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47867 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25850 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29552 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5195 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4112 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/48615 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51487 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5463 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60113 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/54775 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30692 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5585 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52156 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31777 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47937 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51632 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12304 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32858 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/67037 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31524 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12766 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/DerivedSources-input.xcfilelist:
```diff
@@ -687,3 +687,4 @@ $(WEBCORE_PRIVATE_HEADERS_DIR)/ProtectionSpaceBase.serialization.in
 $(WEBCORE_PRIVATE_HEADERS_DIR)/ScrollTypes.serialization.in
 $(WEBCORE_PRIVATE_HEADERS_DIR)/WebGPU.serialization.in
 $(WEBCORE_PRIVATE_HEADERS_DIR)/generate-bindings.pl
+/Applications/Xcode3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.0.Internal.sdk/usr/local/include/WebKitAdditions/TextStyleManager.swift.in
```
This path looks too specific 😅

megangardner replied:
yeah, how did that happen :(

megangardner replied:
That... shouldn't be there at all 🤔

pxlcoder commented on Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h:
```diff
@@ -37,8 +37,6 @@ class WebViewImpl;
 
 @interface WKTextIndicatorStyleManager : NSObject
 
-+ (BOOL)supportsTextIndicatorStyle;
```
Just to confirm: This is being removed as we no longer need to worry about the framework being missing on latest macOS right?

megangardner replied:
Yes, it should always be there/be guarded in other ways, this was for dealing with the soft-linking, etc, while in development.

aprotyas commented on Source/WebKit/DerivedSources-input.xcfilelist:
```diff
@@ -687,3 +687,4 @@ $(WEBCORE_PRIVATE_HEADERS_DIR)/ProtectionSpaceBase.serialization.in
 $(WEBCORE_PRIVATE_HEADERS_DIR)/ScrollTypes.serialization.in
 $(WEBCORE_PRIVATE_HEADERS_DIR)/WebGPU.serialization.in
 $(WEBCORE_PRIVATE_HEADERS_DIR)/generate-bindings.pl
+/Applications/Xcode3.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.Internal.sdk/usr/local/include/WebKitAdditions/TextStyleManager.swift.in
```
Oops, the latest push didn't remove this

megangardner replied:
I deleted this.... I don't know how it came back... or how it got there in the first place...

aprotyas commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -234,7 +234,13 @@ WK_BROWSERENGINEKIT_LDFLAGS[sdk=appletv*] =
+WK_WRITING_TOOLS_UI_LDFLAGS_iphoneos = -framework WritingToolsUI;
+WK_WRITING_TOOLS_UI_LDFLAGS_iphonesimulator = -framework WritingToolsUI;
+WK_WRITING_TOOLS_UI_LDFLAGS_maccatalyst = -framework WritingToolsUI;
+WK_WRITING_TOOLS_UI_LDFLAGS_macosx = -weak_framework WritingToolsUI;
```
I think we need to be conditional about these links to maintain green public EWS, since it still builds against iOS 17-era SDKs.

megangardner replied:
fixed!

rr-codes commented on Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm:
```diff
@@ -23,10 +23,220 @@
-#if USE(APPLE_INTERNAL_SDK)
-#import <WebKitAdditions/WKTextIndicatorStyleManagerAdditions.mm>
-#endif
+#if ENABLE(WRITING_TOOLS_UI) && PLATFORM (MAC)
```
```suggestion
#if ENABLE(WRITING_TOOLS_UI) && PLATFORM(MAC)
```

megangardner committed:
Upstream Writing Tools Animations.
https://bugs.webkit.org/show_bug.cgi?id=275411
rdar://129697392

Reviewed by Richard Robinson.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm:
(-[WKTextIndicatorStyleEffectData initWithEffectID:type:]):
(-[WKTextIndicatorStyleEffectData effectID]):
(-[WKTextIndicatorStyleManager initWithWebViewImpl:]):
(-[WKTextIndicatorStyleManager addTextIndicatorStyleForID:withData:]):
(-[WKTextIndicatorStyleManager removeTextIndicatorStyleForID:]):
(-[WKTextIndicatorStyleManager hasActiveTextIndicatorStyle]):
(-[WKTextIndicatorStyleManager suppressTextIndicatorStyle]):
(-[WKTextIndicatorStyleManager restoreTextIndicatorStyle]):
(-[WKTextIndicatorStyleManager textPreviewsForChunk:completion:]):
(-[WKTextIndicatorStyleManager textPreviewForRect:completion:]):
(-[WKTextIndicatorStyleManager updateIsTextVisible:forChunk:completion:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/TextIndicatorStyle/TextStyleManager.swift: Added.
(currentEffect):
(delegate):
(beginEffect(for:style:)):
(endEffect(for:)):
(TextStyleManager.targetedPreview(for:)):
(TextStyleManager.updateTextChunkVisibilityForAnimation(_:visible:)):
(TextStyleManager.performReplacementAndGeneratePreview(for:effect:animation:)):
(TextStyleManager.replacementEffectDidComplete(_:)):

Canonical link: https://commits.webkit.org/280034@main
